starter-code/routes/authentication.js

```diff
@@ -1,37 +1,66 @@
-const express    = require('express');
-const passport   = require('passport');
-const router     = express.Router();
+const express = require('express');
+const passport = require('passport');
+const multer = require('multer');
+const upload = multer({ dest: './public/uploads/' });
+
+
+const router = express.Router();
 const { ensureLoggedIn, ensureLoggedOut } = require('connect-ensure-login');
 
 router.get('/login', ensureLoggedOut(), (req, res) => {
-    res.render('authentication/login', { message: req.flash('error')});
+  res.render('authentication/login', { message: req.flash('error') });
 });
 
 router.post('/login', ensureLoggedOut(), passport.authenticate('local-login', {
-  successRedirect : '/',
-  failureRedirect : '/login',
-  failureFlash : true
+  successRedirect: '/',
+  failureRedirect: '/login',
+  failureFlash: true
 }));
 
 router.get('/signup', ensureLoggedOut(), (req, res) => {
-    res.render('authentication/signup', { message: req.flash('error')});
+  res.render('authentication/signup', { message: req.flash('error') });
 });
 
-router.post('/signup', ensureLoggedOut(), passport.authenticate('local-signup', {
-  successRedirect : '/',
-  failureRedirect : '/signup',
-  failureFlash : true
-}));
+router.post('/signup', upload.single('photo'), passport.authenticate('local-signup', {
+  successRedirect: '/',
+  failureRedirect: '/signup',
+  failureFlash: true
+}, ((req, res) => {
+  const { username, email, password, avatar } = req.body;
+  const user = {
+    username,
+    email,
+    password,
+    avatar: {
+      name: req.body.name,
+      path: `/uploads/${req.body.filename}`,
+      filename: req.body.filename,
+    }
+  }
+  userInfo.save((err) => {
+    if (err) return res.render('authentication/signup', { error: 'Upload failed' })
```

Comment:
I think there's a problem with the view folder name. The folder is named "auth" and you're looking for "authentication" so we're getting a error in finding the file. Code looks good though